EIP 3549: Metadata Signature Extension (#3542)
ethereum/EIPs: EIPS/eip-3549.md

---

> eip: 3549

Reviewer: ```suggestion
eip: 3542
```

Reviewer: Please update the filename to `eip-3541.md` as well.

title: ERC-721 Metadata Signature Standard

> author: Clay Graham <claytantor@gmail.com>

Reviewer: ```suggestion
author: Clay Graham (@claytantor)
```
We require at least one author list a github handle so they can be tagged for notifications.

type: Standards Track
category: ERC
status: Proposed
created: 2020-04-26

> requires: 165

Reviewer: ```suggestion
requires: 165, 721
```
Is there any reason this wouldn't require EIP-721?

---

## Simple Summary
Add signing to the metadata standard and required fields.

## Abstract

> Recently Tim Schneider, from Artnet posted a great story about how a whitehat hacker managed to trick the blockchain into forging tokens that would forge the Provenance of the originator of the NFT to look as though it was the originating artist.

Reviewer: Please move this anecdote to the motivation.

The **metadata extension** is REQUIRED for ERC-721 smart contracts that require verification using this standard. This allows your smart contract to be interrogated for its name and for details about the assets which your NFTs represent.

```solidity
/// @title ERC-721 Non-Fungible Token Standard, REQUIRED metadata extension
/// @dev See https://eips.ethereum.org/EIPS/eip-721
///  Note: the ERC-165 identifier for this interface is 0x5b5e139f.
interface ERC721Metadata /* is ERC721 */ {
    /// @notice A descriptive name for a collection of NFTs in this contract
    function name() external view returns (string _name);

    /// @notice An abbreviated name for NFTs in this contract
    function symbol() external view returns (string _symbol);

    /// @notice A distinct Uniform Resource Identifier (URI) for a given asset.
    /// @dev Throws if `_tokenId` is not a valid NFT. URIs are defined in RFC
    ///  3986. The URI may point to a JSON file that conforms to the "ERC721
    ///  Metadata JSON Schema".
    function tokenURI(uint256 _tokenId) external view returns (string);

}
```

> This is the "ERC721 Metadata JSON Schema" referenced above.

Reviewer: ```suggestion
This is the "ERC-721 Metadata JSON Schema" referenced above.
```

**Provenance**

Wallets/brokers/auctioneers identified in the motivation section have a strong need to identify which NFTs an owner owns.

It is also identified in the motivation section that being able to identify origin in a manner that provides trust is valuable to the standard. Verified provenance can prove the authenticity of a piece and greatly increase its value.

**Metadata Choices** (metadata extension)

We have required `md5` and `sig` fields in the metadata extension.

Metadata is returned as a string value. Currently this is only usable as calling from `web3`, not from other contracts. This is acceptable because we have not considered a use case where an on-blockchain application would query such information.

## Implementations

0xcert ERC721 -- a reference implementation

- MIT licensed, so you can freely use it for your projects
- Includes test cases
- Active bug bounty, you will be paid if you find errors

> ## References

Reviewer: Please remove this section. It is not valid per EIP-1.
